recipes/qt/5.x.x/conanfile.py

```diff
@@ -168,43 +145,18 @@ def build_requirements(self):
             else:
                 self.tool_requires("winflexbison/2.5.24")
 
-            # Check if a valid python2 is available in PATH or it will failflex
-            # Start by checking if python2 can be found
-            python_exe = tools.which("python2")
-            if not python_exe:
-                # Fall back on regular python
-                python_exe = tools.which("python")
-
-            if not python_exe:
-                msg = ("Python2 must be available in PATH "
-                       "in order to build Qt WebEngine")
-                raise ConanInvalidConfiguration(msg)
-
-            # In any case, check its actual version for compatibility
-            from six import StringIO  # Python 2 and 3 compatible
-            mybuf = StringIO()
-            cmd_v = f"\"{python_exe}\" --version"
-            self.run(cmd_v, output=mybuf)
-            verstr = mybuf.getvalue().strip().split("Python ")[1]
-            if verstr.endswith("+"):
-                verstr = verstr[:-1]
-            version = Version(verstr)
-            # >= 2.7.5 & < 3
-            v_min = "2.7.5"
-            v_max = "3.0.0"
-            if (version >= v_min) and (version < v_max):
-                msg = ("Found valid Python 2 required for QtWebengine:"
-                       f" version={mybuf.getvalue()}, path={python_exe}")
-                self.output.success(msg)
-            else:
-                msg = (f"Found Python 2 in path, but with invalid version {verstr}"
-                       f" (QtWebEngine requires >= {v_min} & < {v_max})\n"
-                       "If you have both Python 2 and 3 installed, copy the python 2 executable to"
-                       "python2(.exe)")
-                raise ConanInvalidConfiguration(msg)
-
         if self.options.qtwayland:
             self.tool_requires("wayland/1.21.0")
+        if self.options.with_doubleconversion:
+            self.tool_requires("double-conversion/3.2.1")
+        if self.options.get_safe("with_icu", False):
+            self.tool_requires("icu/72.1")
+        if self.options.with_pcre2:
+            self.tool_requires("pcre2/10.42")
+        if self.options.with_zstd:
+            self.tool_requires("zstd/1.5.4")
+        if self.options.with_glib:
+            self.tool_requires("glib/2.75.3")
```

Comment:
I don't understand. What does it mean to have double-conversion in build context? There is no executable.

Reply:
They are used by executables built by qt and called during qt build: https://c3i.jfrog.io/c3i/misc-v2/logs/pr/16404/18-linux-gcc/qt/5.15.6//4789f98e3abda2a0a151b8cbd031ce2887b7aada-build.txt

Reply:
It's a broken recipe logic since these executables are linked to requirements flavors, not build requirements. It may work inadvertently when requirements & build requirements are exactly the same configuration (same shared flavor basically) like in conan v1 pipeline where all requirements & build requirements are static.

I've opened https://github.com/conan-io/conan-center-index/pull/17030